sdk/storage/storage-blob/src/pollers/BlobStartCopyFromUrlPoller.ts

```diff
@@ -0,0 +1,187 @@
+import { delay } from "@azure/core-http";
+import { PollOperation, PollOperationState, Poller } from "@azure/core-lro";
+import { BlobStartCopyFromURLResponse } from "../generated/src/models";
+import { BlobClient, BlobStartCopyFromURLOptions } from "../BlobClient";
+
+/**
+ * Defines the operations needed from a BlobClient.
+ */
+export type CopyPollerBlobClient = Pick<BlobClient, "abortCopyFromURL" | "getProperties"> & {
+  startCopyFromURL(
+    copySource: string,
+    options?: BlobStartCopyFromURLOptions
+  ): Promise<BlobStartCopyFromURLResponse>;
+};
+
+/**
+ * The state used by the PollOperation.
+ */
+export interface BlobBeginCopyFromUrlPollState
+  extends PollOperationState<BlobStartCopyFromURLResponse> {
+  blobClient: CopyPollerBlobClient;
+  copyId?: string;
+  copyProgress?: string;
+  copySource: string;
+  startCopyFromURLOptions?: BlobStartCopyFromURLOptions;
+}
+
+/**
+ * The PollOperation responsible for:
+ *  - performing the initial startCopyFromURL
+ *  - checking the copy status via getProperties
+ *  - cancellation via abortCopyFromURL
+ */
+export interface BlobBeginCopyFromURLPollOperation
+  extends PollOperation<BlobBeginCopyFromUrlPollState, BlobStartCopyFromURLResponse> {}
+
+/**
+ *
+ */
+export interface BlobBeginCopyFromUrlPollerOptions {
+  blobClient: CopyPollerBlobClient;
+  copySource: string;
+  intervalInMs?: number;
+  onProgress?: (state: BlobBeginCopyFromUrlPollState) => void;
+  resumeFrom?: string;
+  startCopyFromURLOptions?: BlobStartCopyFromURLOptions;
+}
+
+/**
+ *
+ */
+export class BlobBeginCopyFromUrlPoller extends Poller<
```

Comment:
>BlobBeginCopyFromUrlPoller [](start = 13, length = 26)

JSdoc

Reply:
The user won't see this, it's an internal class but I'll add some comments to it.
We expose an interface for the return value of `beginCopyFromURL`